website/static/js/registrationUtils.js

```diff
@@ -260,10 +260,15 @@ var MetaSchema = function(params) {
     self.fulfills = params.fulfills || [];
     self.messages = params.messages || {};
 
+    var count = 0;
     $.each(self.schema.pages, function(i, page) {
         var mapped = {};
         $.each(page.questions, function(qid, question) {
-            mapped[qid] = new Question(question, qid);
+            // This is necessary now since page.questions is a list
```

Comment:
As for question IDs: if I read this correctly, the question is already represented as an object. Instead of storing the QID outside the question (in the larger hash of questions), why not add another key? 

(Disclaimer: remember, I don't know what all the fields in the schema are. Grain of salt, etc etc)

Old:
```
{
    "type": "string",
    "format": "text",
    "title": "The confidence interval of the replication effect size is",
    "nav": "Confidence interval"
}
```

New:
```
{
    "qid": "q42",
    "type": "string",
    "format": "text",
    "title": "The confidence interval of the replication effect size is",
    "nav": "Confidence interval"
}
```